app/pages/lab-pages-editor/helpers/getNewStepKey.js

```diff
@@ -0,0 +1,20 @@
+/*
+Searches the Workflow's existing Steps and finds the next available Step Key.
+- Goes through workflow.steps and finds steps in the format of P0, P1, P2, etc
+- Finds the LARGEST/HIGHEST key and returns one higher.
+- e.g. if given [P0, P1, P5], this will return P6.
+- Return P0 by default.
+ */
+
+import { STEP_KEY_PREFIX } from './constants.js';
+import convertKeyToIndex from './convertKeyToIndex.js';
+
+export default function getNewStepKey(steps = []) {
+  let newIndex = 0;
+  const stepKeys = steps.map((step) => (step[0] || '')).filter((step) => (step.length > 0));
+  stepKeys.forEach((stepKey) => {
+    const index = convertKeyToIndex(stepKey);
+    newIndex = (newIndex <= index) ? index + 1 : newIndex;
+  });
+  return `${STEP_KEY_PREFIX}${newIndex}`;
```

Comment:
Live transcription workflows have step keys like `S0` and `S1`. See Davy Notebooks or PRINT for example. Will they work with this function?

Reply:
Yeah, dealing with existing workflows (and a potential "autocleanup/autoconvert" function) is something I'm still sweating about. 😓 

To answer the question, with the earlier version of convertKeyToIndex, if an existing WF has `steps = S0, S1`, this function should return `P0`, so the WF would have an odd mix of `steps = S0, S1, P0`, but should still ~~technically~~ _teeechnically...?_ work.

With the newer version of convertKeyToIndex, this would now return `P3`, so we'd _still_ have an odd mix of `steps = S0, S1, P2`. 😞 Either case, this is something on my To Test list, once I get step chaining implemented (i.e. properly get that `.next` working) in the next (or more like _next next)_ PR.

Reply:
Oh! Forgot to add context for this: with the switch of the user-facing nomenclature to favour "Page" over "Step", we wanted to see if `P0` was possible. This _isn't_ a non-negotiable parameter as far as I know, so if using P becomes a huge Pain, I'll just switch that STEP_KEY_PREFIX back to `S`

(Please don't ask me about the topic starting pages/steps as P1 instead of P0.)

Reply:
That's a good point. I think some of the monorepo tests start workflows at `S1`.

Reply:
`next` should be optional. Maps are iterables, in the order that items were added to the map. A linear workflow should be able to default to iterating through the steps map in order. 

`next` is needed for decision trees, where a workflow has more than one final step, or where steps branch.